File: 0448-find-all-numbers-disappeared-in-an-array/0448-find-all-numbers-disappeared-in-an-array.java
class Solution {
    public List<Integer> findDisappearedNumbers(int[] nums) {
        int i = 0;
        while (i < nums.length){
            int correct = nums[i]-1;
            if(nums[i] != nums[correct]){
                swap(nums, i, correct);
               
            }else{
                 i++;
                
            }
            
        }
        List<Integer> ans = new ArrayList<>(); 
        for (int a=0; a < nums.length; a++){
            if (nums[a] != a+1){
                ans.add(a+1);
                
            }
        }
        return ans;
        
        
        
        
    }
    public void swap(int[] nums, int first, int second){
        int temp = nums[first];
        nums[first] = nums[second];
        nums[second] = temp;
    }
}

File: 0448-find-all-numbers-disappeared-in-an-array/0448-find-all-numbers-disappeared-in-an-array.py
class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
            
        def swap(arr, i, j):
            arr[i], arr[j] = arr[j], arr[i]
        i = 0
        while i<len(nums):
            correct= nums[i]-1
            if nums[i] == nums[correct]:
                i+=1
            else:
                swap(nums, i, correct)
        
        result=[]
        for i in range(len(nums)):
            if nums[i] != i+1:
                result.append(i+1)
        return result 
            
    